google/cloud/access_token_generator.cc

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/access_token_generator.h"
+#include "google/cloud/internal/unified_rest_credentials.h"
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+class AccessTokenGeneratorImpl : public AccessTokenGenerator {
+ public:
+  explicit AccessTokenGeneratorImpl(
+      std::shared_ptr<google::cloud::oauth2_internal::Credentials> impl)
+      : impl_(std::move(impl)) {}
+  ~AccessTokenGeneratorImpl() override = default;
+
+  StatusOr<AccessToken> GetToken() override {
+    return impl_->GetToken(std::chrono::system_clock::now());
+  }
+
+ private:
+  std::shared_ptr<google::cloud::oauth2_internal::Credentials> impl_;
```

Comment:
Is `impl_` the right name, versus, say `creds_`?

Reply:
I can argue both sides of that.  And sound like I believe both of them too!  Changed to `creds_`.